recipes-kernel/accton-csp7551-mods/accton-csp7551-mods.bb

```diff
@@ -0,0 +1,19 @@
+SUMMARY = "CSP 7551 platform modules"
+LICENSE = "GPL-2.0-only"
```

Comment:
Your COPYING, as well as `accton_i2c_cpld.c` and `x86-64-accton-csp7551-sfp.c` say:

>  This program is free software; you can redistribute it and/or modify
>  it under the terms of the GNU General Public License as published by
>  the Free Software Foundation; either version 2 of the License, or
>  (at your option) **any later** version.

The `fpga_driver.c` is missing a license header and only has a `MODULE_LICENSE("GPL");`, which can mean anything. but I guess we can assume they mean the same license.

Reply:
Considering that the driver is coming from a public SONiC repo, I believe that is a fair assumption to make.

Reply:
I guess my point didn't come across: The actual license says GPL 2.0 or later, not GPL 2.0 only. The LICENSE in the recipe is wrong (and needs to be `GPL-2.0-or-later` or so).

Reply:
The GPL license at the top of the file in `accton_i2c_cpld.c` explicitly says "either version 2 of the License, or (at your option) any later version", and the GPL license at the top `x86-64-accton-csp7551-sfp.c` says "either version 2 of the License, or (at your option) any later version". Both of these contain the "or later" wording, so they are by definition `GPL-2.0-or-later`.

See e.g. for comparison [x86-64-delta-ag7648-cpld-mux-1.c](https://github.com/opencomputeproject/OpenNetworkLinux/blob/master/packages/platforms/delta/x86-64/ag7648/modules/builds/x86-64-delta-ag7648-cpld-mux-1.c) which explicitly states it is version 2.